src/App.tsx

```diff
@@ -19,13 +27,15 @@ export const App: React.FC = () => {
                 type="text"
                 id="search-query"
                 className="input"
+                value={query}
                 placeholder="Type search word"
+                onChange={event => setQuery(event.target.value.toLowerCase())}
               />
             </div>
           </div>
         </div>
 
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={query === '' ? moviesFromServer : visibleMovies} />
```

Comment:
you can remove the ternary operator
visibleMovies is always correct array even if `query === ''`